PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-springcloud/src/main/java/com/microsoft/azure/toolkit/intellij/springcloud/properties/SpringCloudAppPropertiesEditor.java

```diff
@@ -156,27 +157,14 @@ private void refreshToolbar() {
                 AzureMessager.getMessager().warning(String.format("App(%s) has no active deployment", this.app.getName()), null);
             }
             AzureTaskManager.getInstance().runLater(() -> {
-                switch (status.toUpperCase()) {
-                    case "STOPPED":
-                    case "RUNNING":
-                        this.setEnabled(true);
-                        this.resetButton.setVisible(modified);
-                        this.saveButton.setEnabled(modified);
-                        this.startButton.setEnabled(this.app.isStartable());
-                        this.stopButton.setEnabled(this.app.isStoppable());
-                        this.restartButton.setEnabled(this.app.isStoppable());
-                        break;
-                    case "FAILED":
-                    case "INACTIVE":
-                    case "ALLOCATING":
-                    case "UPGRADING":
-                    case "COMPILING":
-                    case "UNKNOWN":
-                        this.setEnabled(false);
-                        this.deleteButton.setEnabled(true);
-                        break;
-                    default:
-                }
+                final AzResourceBase.FormalStatus formalStatus = this.app.getFormalStatus();
+                this.setEnabled(formalStatus.isRunning() || formalStatus.isStopped());
+                this.resetButton.setVisible(modified && !formalStatus.isWriting());
```

Comment:
change to `(formalStatus.isRunning() || formalStatus.isStopped()) && modified` to keep consistent with enabled?